main_test.go

```diff
@@ -72,48 +72,125 @@ func Test_filterFeatures(t *testing.T) {
 			expectedError:    nil,
 		},
 		{
-			title: "Every field which defines populated feature is populated",
+			title: "Every field which defines populated feature is populated for github",
 			planConfig: types.Plan{
-				Github: types.Github{
-					AppID:          "example ID",
-					PrivateKeyFile: "~/home/private_key.pem",
+				SCM: types.GitHubSCM,
+				TLS: true,
+				TLSConfig: types.TLSConfig{
```

Comment:
I think TLS should not be related to the scm option. Why does it replace the GitHub specific data?

Reply:
Well before you added the `scm: gitlab` there was only `scm: github` so I could not decide by that I picked if the object `GitHub` is populated then the filter should be added. ATM I changed the logic so if `scm: gitlab` then filter `scm_gitlab` is enabled. TLS comes with this test case just to check more combinations.